src/map/status_effect_container.cpp

```diff
@@ -792,6 +792,16 @@ void CStatusEffectContainer::DelStatusEffectsByFlag(uint32 flag, bool silent)
     {
         if (PStatusEffect->HasEffectFlag(flag))
         {
+            // If this is an NM/Mob Nightmare sleep, it can be removed explictly by a cure
+            // see mobskills/nightmare.lua for full explanation
```

Comment:
What is the mechanism that cure uses to bypass this check?

Reply:
This block is only for deleting by `damage` flag. A cure would wake via a different means